src/coreclr/src/System.Private.CoreLib/src/System/Runtime/CompilerServices/CastHelpers.cs

```diff
@@ -227,26 +227,6 @@ private static CastResult TryGet(nuint source, nuint target)
                             goto done;
                     }
                     while ((i += 4) < interfaceCount);
-
-                    // REVIEW FYI: Also tried the following (vectorizing the lookup).
-                    //
-                    //             It improved the worst case scenarios (10-20%), since we can get through the whole list faster.
-                    //             but best/common cases got slower (also 10-20%). I am guessing that superscalar CPU is very good at
-                    //             linear/speculative compares, and without any math vectorization does not help that much.
-                    //
-
-                    //var toTypeV = Vector256.Create((nuint)toTypeHnd);
```

Comment:
One interesting thing I tried here - vectorizing interface lookup via `Avx2` - by doing 4 compares at a time. It did not help enough though. Speculative CPU seems to be good at these compares. Probably does 4 at a time already.

It was an interesting try though.